tests/NCronJob.Tests/RuntimeJobRegistryTests.cs

```diff
@@ -214,6 +256,54 @@ public async Task ShouldDisableJob()
         jobFinished.ShouldBeFalse();
     }
 
+    [Fact]
+    public async Task DisablingAndEnablingByJobTypeAccountsForAllJobs()
+    {
+        ServiceCollection.AddNCronJob(s => s.AddJob<SimpleJob>());
+        ServiceCollection.AddNCronJob(s => s.AddJob<SimpleJob>(p => p.WithCronExpression(AtMinute2)));
+
+        var provider = CreateServiceProvider();
+        await provider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
+        var registry = provider.GetRequiredService<IRuntimeJobRegistry>();
+
+        var jobRegistry = provider.GetRequiredService<JobRegistry>();
+        var jobs = jobRegistry.FindAllJobDefinition(typeof(SimpleJob));
+        Assert.Equal(2, jobs.Count);
+
+        Assert.All(jobs, j =>
+        {
+            if (j.CronExpression is null)
+            {
+                return;
+            }
+
+            Assert.NotEqual(RuntimeJobRegistry.TheThirtyFirstOfFebruary, j.CronExpression);
+        });
+
+        registry.DisableJob<SimpleJob>();
+
+        jobs = jobRegistry.FindAllJobDefinition(typeof(SimpleJob));
+        Assert.Equal(2, jobs.Count);
+
+        Assert.All(jobs, j =>
+        {
+            Assert.Equal(RuntimeJobRegistry.TheThirtyFirstOfFebruary, j.CronExpression);
+        });
+
+        registry.EnableJob<SimpleJob>();
+
+        jobs = jobRegistry.FindAllJobDefinition(typeof(SimpleJob));
+        Assert.Equal(2, jobs.Count);
+
+        Assert.All(jobs, j =>
+        {
+            Assert.NotEqual(RuntimeJobRegistry.TheThirtyFirstOfFebruary, j.CronExpression);
```

Comment:
Hmmm.... This assertion is not useful as it's covered by the two following ones.

To be dropped

Reply:
Dropped